### PR TITLE
Fire delete to partners on a Delete Setting Event

### PR DESCRIPTION
**Pull request by rtewalt**
| Status  | Type  | Migrations
| :---: | :---: | :---: |
| Ready | Feature | No |

## Description

When a Delete Setting event is sent through the prism SNS, a DELETE call will be fired to the partner to delete the association for that setting from their systems.

## Other changes

Refactor tests to separate update from delete functional tests.

## Related Tickets

List related Clubhouse tickets:

* [Sync product association information to partners](https://app.clubhouse.io/shipt/story/48704/sync-product-association-information-to-partners)

## Todos

_Fill out or remove list items as needed_

- [x] Tests
- [x] Documentation

## Deploy Notes

N/A

### CODE REVIEW

File: internal/app/syncproduct/cmd/handler.go [hunk 1]
```diff
@@ -61,6 +63,23 @@ func (cmd *PrismDeleteSettingCommand) Handle(request events.SNSEvent) error {
...
+		StoreID:    cmd.Data.StoreID,
+		MetroID:    cmd.Data.MetroID,
+		LocationID: cmd.Data.LocationID,
+		ProductID:  cmd.Data.ProductID,
+	}
+
+	err = webhook.DeleteAssociationByContext(context)
+
+	if err != nil {
+		return errors.Wrapf(err, "Could not DeleteAssociationByContext %v from %v", context, cmd)
```
Comment (brianseitel) [inline internal/app/syncproduct/cmd/handler.go]: 🔥

Comment (rtewalt) [reply internal/app/syncproduct/cmd/handler.go]: is that a good thing? 🤔

Comment (rtewalt) [reply internal/app/syncproduct/cmd/handler.go]: i suppose 🔥 != :dumpster_fire:

Comment (brianseitel) [reply internal/app/syncproduct/cmd/handler.go]: haha, i think the use of `errors.Wrapf` is cool :D

**Comment by rtewalt** [comment]
retest this please

**Pull request merged**